kimchi/src/circuits/polynomials/generic.rs

```diff
@@ -104,173 +330,165 @@ impl<F: FftField + SquareRootField> ConstraintSystem<F> {
         generic_zeta: F,
     ) -> Evaluations<F, D<F>> {
         let d1 = self.domain.d1;
-
-        let (scalars, alpha) = Self::gnrc_scalars(alphas, w_zeta, generic_zeta);
-
         let n = d1.size as usize;
 
+        // get scalars
+        let scalars = Self::gnrc_scalars(alphas, w_zeta, generic_zeta);
+
+        //
         let mut res = Evaluations::from_vec_and_domain(vec![F::zero(); n], d1);
 
         let scale = self.coefficients8[0].evals.len() / n;
 
-        // w[0](zeta) * qwm[0] + w[1](zeta) * qwm[1] + w[2](zeta) * qwm[2]
-        self.coefficients8
-            .iter()
-            .zip(scalars[1..].iter())
-            .for_each(|(q, s)| {
-                res.evals.par_iter_mut().enumerate().for_each(|(i, e)| {
-                    *e += *s * q.evals[scale * i];
-                });
+        let coeffs = self.coefficients8.iter();
+        for (scalar, coeff) in scalars.into_iter().zip(coeffs) {
+            res.evals.par_iter_mut().enumerate().for_each(|(i, e)| {
+                *e += scalar * coeff[scale * i];
             });
-
-        // multiplication
-        let s = scalars[0];
-        res.evals.par_iter_mut().enumerate().for_each(|(i, e)| {
-            *e += s * self.coefficients8[MUL_COEFF][scale * i];
-        });
-
-        // constant selector
-        let alpha_generic_zeta = alpha * generic_zeta;
-        res.evals.par_iter_mut().enumerate().for_each(|(i, e)| {
-            *e += alpha_generic_zeta * self.coefficients8[CONSTANT_COEFF][scale * i];
-        });
+        }
 
         // l * qwm[0] + r * qwm[1] + o * qwm[2] + l * r * qmm + qc
         res
     }
 
-    /// function to verify the generic polynomials with a witness
+    /// Function to verify the generic polynomials with a witness.
```

Comment:
There are 2 `verify_generic` functions in this module; perhaps this one should be `verify_generic_poly` and/or the other should be `verify_generic_witness`?

Reply:
I've been wondering if we really need both actually